src/network/gameclient.cc

```diff
@@ -1217,6 +1212,15 @@ void GameClient::disconnect(const std::string& reason,
 			    (arg.empty() ? NetworkGamingMessages::get_message(reason) :
 			                   NetworkGamingMessages::get_message(reason, arg)))
 			      .str());
+		}
+	}
+
+	// TODO(Klaus Halfmann): Some of the modal windows are now handled by unique_ptr resulting in a
+	// double free.
+	if (d->modal) {
+		if (d->modal->is_modal()) {
+			d->modal->end_modal<FsMenu::MenuTarget>(FsMenu::MenuTarget::kBack);
+		} else {
 			d->modal->die();
 		}
```

Comment:
```suggestion
		d->modal->die();
```
This can be simplified now because `die()` for modal panels simply calls `end_modal(kBack)`.